src/k8s/pkg/client/kubernetes/status.go

```diff
@@ -20,6 +21,25 @@ func (c *Client) WaitKubernetesEndpointAvailable(ctx context.Context) error {
 	})
 }
 
+// IsClusterReady pings /readyz endpoint and only reports ready if status code == 200.
+func (c *Client) IsClusterReady(ctx context.Context) bool {
```

Comment:
`IsClusterReady()` is probably not the right name for this. Perhaps `CheckReadyz() error` is preferred. Return the error and let users handle it, or return `nil` for success. That can reduce the need to log the results as well.

Reply:
I'm not sure I understand "querying for the endpoints" correctly, are we going to do something like this? :
```go
endpoint, err := c.CoreV1().Endpoints("default").Get(ctx, "kubernetes", metav1.GetOptions{})
```
in which case the name for this function should not be `CheckReadyz`, right?

Reply:
right, pick a name that communicates what the function does